### PR TITLE
Merge neo4j-wrapper.conf into neo4j.conf

### PR DESCRIPTION
**Pull request by spacecowboy**
All options should be specified in the single `neo4j.conf` file.

For backwards compatible reasons we will still load settings from `neo4j-wrapper.conf` if they do not exist in `neo4j.conf`, but we should remove all mentions of this file.

changelog: [configuration] Merged `neo4j-wrapper.conf` and `neo4j.conf` into a single configuration file. `neo4j-wrapper.conf` is still parsed if it exists for backward compatibility's sake.

### CODE REVIEW

File: community/common/src/main/java/org/neo4j/kernel/api/exceptions/Status.java [hunk 1]
```diff
@@ -426,14 +426,14 @@ public Code code()
...
-                ".conf' or, if you are using Neo4j Desktop, found through the user interface) or if you are running an embedded " +
+                "'dbms.memory.heap.max_size' in the process configuration (normally in 'conf/neo4j.conf' or, if you " +
+                "you are using Neo4j Desktop, found through the user interface) or if you are running an embedded " +
                 "installation increase the heap by using '-Xmx' command line flag, and then restart the database." ),
         StackOverFlowError( TransientError,
                 "There is not enough stack size to perform the current task. This is generally considered to be a " +
                 "database error, so please contact Neo4j support. You could try increasing the stack size: " +
                 "for example to set the stack size to 2M, add `dbms.jvm.additional=-Xss2M' to " +
-                "in the process wrapper configuration (normally in 'conf/neo4j-wrapper.conf' or, if you are using " +
+                "in the process configuration (normally in 'conf/neo4j.conf' or, if you are using " +
```
Comment (benbc) [inline community/common/src/main/java/org/neo4j/kernel/api/exceptions/Status.java]: I don't think that the phrase "process configuration" makes sense here or above. How about "Neo4j configuration"?

File: manual/server-examples/src/docs/dev/index.asciidoc [hunk 2]
```diff
@@ -15,7 +15,8 @@ When running your own code and Neo4j in the same JVM, there are a few things you
 * Don't use internal Neo4j APIs.
   They may change in future releases, and they might not always work as intended in the context of your code.
 * If possible, avoid using Java object serialization or reflection in your code or in any runtime dependency that you use.
-  Otherwise, if you cannot avoid using Java object serialization and reflection, then ensure that the `-XX:+TrustFinalNonStaticFields` JVM flag is disabled in `neo4j-wrapper.conf`.
+  Otherwise, if you cannot avoid using Java object serialization and reflection, then ensure that the
+  `-XX:+TrustFinalNonStaticFields` JVM flag is disabled in `neo4j.conf`.
```
Comment (benbc) [inline manual/server-examples/src/docs/dev/index.asciidoc]: This sentence should be on a single line.

File: packaging/standalone/pom.xml [hunk 3]
```diff
@@ -40,7 +40,6 @@
     <!-- Document substitution properties are in document.properties file -->
 
     <!-- other filter properties -->
-    <wrapper.conf>conf/neo4j-wrapper.conf</wrapper.conf>
```
Comment (benbc) [inline packaging/standalone/pom.xml:43]: Was this not used anywhere?

File: packaging/standalone/src/tests/shell-scripts/test-java-arguments.sh [hunk 4]
```diff
@@ -13,13 +13,19 @@ for run_command in run_console run_daemon; do
...
 
-  test_expect_success "should add additional options" "
+  test_expect_success "should add additional options from wrapper conf" "
     set_config 'dbms.jvm.additional' '-XX:+UseG1GC', neo4j-wrapper.conf &&
     ${run_command} &&
     test_expect_java_arg '-XX:+UseG1GC'
   "
 
-  test_expect_success "should set heap size constraints" "
+    test_expect_success "should add additional options" "
```
Comment (benbc) [inline packaging/standalone/src/tests/shell-scripts/test-java-arguments.sh]: Indentation wrong?

**Comment by benbc** [comment]
I think that the new bits of `neo4j.conf` would be better at the end of the file, rather than before the "misc" section.

**Comment by benbc** [comment]
Apart from the `dbms.memory.*` settings which should go right to the top, next to the pagecache size setting.

File: manual/server-examples/src/docs/dev/index.asciidoc (hunk 2 above)
Comment (spacecowboy) [inline manual/server-examples/src/docs/dev/index.asciidoc]: How come? Asciidoc will render it as a single line anyway

File: packaging/standalone/pom.xml (hunk 3 above)
Comment (spacecowboy) [inline packaging/standalone/pom.xml:43]: Not that I could find

File: manual/server-examples/src/docs/dev/index.asciidoc (hunk 2 above)
Comment (benbc) [inline manual/server-examples/src/docs/dev/index.asciidoc]: Docs coding standard. Make the diffs easier to read.

**Pull request merged**